packages/mesos/extra/patches/0005-Changed-sandbox-mode-to-0755-in-docker-containerizer.patch

```diff
@@ -0,0 +1,45 @@
+From d2ee1b4542232042dd6fa532484f2082f96c0163 Mon Sep 17 00:00:00 2001
+From: Gilbert Song <songzihao1990@gmail.com>
+Date: Fri, 23 Nov 2018 10:54:58 +0800
+Subject: [PATCH] Changed sandbox mode to 0755 in docker containerizer
+ container launch.
+
+Starting from Mesos 1.6.0, sandbox permission was changed from 0755
+to 0750 to narrow access from non-task users. This is a semantic
+change for docker containerizer launched container which honors a
+default user from the docker image. See MESOS-8332 and DCOS_OSS-4415
+for details.
+
+This commit is used to patch DC/OS and allows DC/OS users to have a
+deprecation cycle for this semantic change. This commit should be
+removed in DC/OS 1.15.
+---
+ src/slave/containerizer/docker.cpp | 12 ++++++++++++
+ 1 file changed, 12 insertions(+)
+
+diff --git a/src/slave/containerizer/docker.cpp b/src/slave/containerizer/docker.cpp
+index fc29e20..6f52150 100644
+--- a/src/slave/containerizer/docker.cpp
++++ b/src/slave/containerizer/docker.cpp
+@@ -1167,6 +1167,18 @@ Future<Containerizer::LaunchResult> DockerContainerizerProcess::launch(
+     return Containerizer::LaunchResult::NOT_SUPPORTED;
+   }
+ 
++  // TODO(gilbert): This is a patch for DC/OS, in order to work
++  // around the sandbox permission change in Mesos 1.6.0, which
++  // is not backward compatible in DC/OS. Please see MESOS-8332
++  // and DCOS_OSS-4415 for details. Remove this workaround after
++  // a deprecation cycle in DC/OS 1.15.
++  Try<Nothing> chmod = os::chmod(containerConfig.directory(), 0755);
```

Comment:
Why don't we simply change '0750' to '0755' permissions in https://github.com/apache/mesos/blob/8c6e0e570e970a1fa6e1b3085373c6b84cb4a81d/src/slave/paths.cpp#L801 ?

Reply:
Because 0750 is the right semantic to go for the future and this semantic change only impact docker containerizer launching containers with an image that honor a default image user. UCR uses task's user and image user is not supported yet, so UCR is fine.